spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/pom.xml

```diff
@@ -65,4 +83,97 @@
       </plugin>
     </plugins>
   </build>
+
+  <profiles>
+    <profile>
+      <id>native</id>
+      <properties>
+        <integration-test.tags.include>native</integration-test.tags.include>
+        <integration-test.tags.exclude/>
+        <!-- Customize Spring Boot classifier, useful in order to avoid a clash between Spring Boot repackaging and native-image-maven-plugin -->
+        <classifier/>
+        <!-- Customize native-image arguments with this property -->
+        <native.build.args>
+          --enable-http
+          --enable-https
+        </native.build.args>
+
+        <!-- `tiny` builder allows small footprint and reduced surface attack, you can also use `base` (the default) or `full` builders to have more tooling available in the image for an improved developer experience -->
+        <!-- TODO Bring back "paketobuildpacks/builder:tiny" before 0.10.0 release -->
+        <builder>dmikusa/graalvm-tiny</builder>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.springframework.experimental</groupId>
+          <artifactId>spring-native</artifactId>
+          <version>${spring-native.version}</version>
+        </dependency>
+        <dependency>
+          <groupId>com.google.cloud</groupId>
+          <artifactId>google-cloud-graalvm-support</artifactId>
+          <version>${google-cloud-graalvm-support.version}</version>
+        </dependency>
+        <dependency>
+          <groupId>org.springframework</groupId>
+          <artifactId>spring-context-indexer</artifactId>
+          <optional>true</optional>
+        </dependency>
+      </dependencies>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.springframework.experimental</groupId>
+            <artifactId>spring-aot-maven-plugin</artifactId>
+            <version>${spring-native.version}</version>
+            <dependencies>
+              <dependency>
+                <groupId>com.google.cloud</groupId>
+                <artifactId>spring-cloud-gcp-native-support</artifactId>
+                <version>${project.version}</version>
+              </dependency>
+            </dependencies>
+            <executions>
+              <execution>
+                <id>test-generate</id>
+                <goals>
+                  <goal>test-generate</goal>
+                </goals>
+              </execution>
+              <execution>
+                <id>generate</id>
+                <goals>
+                  <goal>generate</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>build-docker-image-before-integration-tests</id>
```

Comment:
Were there any blockers from getting the sample to be built on machine over doing a docker build? Mostly asking to figure out how to best incorporate this in our CI.

Reply:
graalvm is needed to build the project, but I think you have a fair point here. We can get rid of docker, and just run the native executable here 🤔

Would that be the preferred solution?

Reply:
Gotcha I see. No worries, I think what you have is fine.

I'll add a commit to your branch to try to incorporate this in our Github Actions and see if that would work.

I'm thinking users probably benefit from doing things in docker/buildpacks, and then for our CI maybe we try to just build local on the CI machines.